protocols/imap/src/main/java/org/apache/james/imap/decode/parser/AbstractSelectionCommandParser.java

```diff
@@ -129,7 +130,19 @@ protected ImapMessage decode(ImapRequestLineReader request, Tag tag, ImapSession
         request.eol();
         return createRequest(mailboxName, condstore, lastKnownUidValidity, knownModSeq, uidSet, knownUidSet, knownSequenceSet, tag);
     }
-    
+
+    private UidValidity sanitizeUidValidity(long uidValidityAsNumber) {
```

Comment:
I don't understand why it's needed

Reply:
What if the client provides an invalid Uid validity that we previously returned to him?

Reply:
So the plan is the generate a valid uidvalidity that won't match the one store in DB and it will force synchronization?
I don't like have a random value used as a fake client-provided one.
I'm back with my Null Object Pattern: we shouldn't lie to the code reader so it should be clear it's an "invalid-user-provided-uidvalidity-that-we-have-to-accept-because-of-legacy-values"